core/src/services/split-view.js

```diff
@@ -0,0 +1,215 @@
+      };
+    }
+  }
+
+  enforceTreshHolds(top, bottom, thresholdTop, thresholdBottom) {
+    if (top <= thresholdTop) {
+      top = thresholdTop;
+      bottom = window.innerHeight - thresholdTop;
+    } else if (bottom <= thresholdBottom) {
+      top = window.innerHeight - thresholdBottom;
+      bottom = thresholdBottom;
+    }
+    return { top, bottom };
+  }
+
+  setDeep(comp, key, value) {
+    comp.set({
+      [key]: Object.assign({}, comp.get()[key], value)
+    });
+  }
+
+  open(comp, nodepath, settings) {
+    const mfSplitView = {
+      displayed: true,
+      collapsed: false, // settings.collapsed === true, // TODO: separate ticket
+      nodepath,
+      settings
+    };
+
+    this.splitViewValues = this.calculateInitialValues(
+      mfSplitView.settings && mfSplitView.settings.size,
+      GenericHelpers.getRightContentHeight()
+    );
+
+    this.sendClientEvent('internal', {
+      exists: true,
+      size: this.splitViewValues.percent,
+      collapsed: mfSplitView.collapsed
+    });
+    comp.set({ mfSplitView, splitViewValues: this.splitViewValues });
+  }
+
+  expand(comp) {
+    if (comp.get().splitViewIframe) {
+      this.sendClientEvent('internal', {
+        exists: true,
+        size: this.storedSplitViewValues.percent,
+        collapsed: false
+      });
+      this.sendClientEvent('expand');
+
+      this.setDeep(comp, 'mfSplitView', {
+        displayed: true,
+        collapsed: false
+      });
+
+      this.getContainer().style.top = `${this.storedSplitViewValues.top}px`;
+      Iframe.getIframeContainer().style.paddingBottom = `${
+        this.storedSplitViewValues.bottom
+      }px`;
+      this.getDragger().style.top = `${this.storedSplitViewValues.top}px`;
+      this.storedSplitViewValues = undefined;
+    }
+  }
+
+  collapse(comp) {
+    if (comp.get().splitViewIframe) {
+      comp
+        .getUnsavedChangesModalPromise(comp.get().splitViewIframe.contentWindow)
+        .then(() => {
+          this.sendClientEvent('internal', {
+            exists: true,
+            size: this.splitViewValues.percent,
+            collapsed: true
+          });
+          this.sendClientEvent('collapse');
+          this.setDeep(comp, 'mfSplitView', {
+            displayed: true,
+            collapsed: true
+          });
+          this.storedSplitViewValues = Object.assign({}, this.splitViewValues);
+          this.getContainer().style.top = '';
+          Iframe.getIframeContainer().style.paddingBottom = '';
+        });
+    }
+  }
+
+  closeSplitView(comp) {
+    if (comp.get().splitViewIframe) {
+      comp
+        .getUnsavedChangesModalPromise(comp.get().splitViewIframe.contentWindow)
+        .then(() => {
+          this.setDeep(comp, 'mfSplitView', {
+            displayed: false,
+            collapsed: false
+          });
+          Iframe.getIframeContainer().style.paddingBottom = '';
+          this.sendClientEvent('close');
+        });
+    }
+  }
+
+  sendClientEvent(name, data) {
```

Comment:
shall we check if `name` has a proper value and in case not display a console warning instead of sending the message?